packages/core/lib/src/models/media.dart

```diff
@@ -0,0 +1,43 @@
+/// Enum for the type of media in a tweet
+enum MediaType {
+  /// The media is a video
+  video,
+
+  /// The media is a GIF
+  gif,
+
+  /// The media is a single image
+  photo,
+}
+
+/// Tweet Media object
+class Media {
+  /// Creates a new instance of [Media]
+  const Media({
+    required this.key,
+    required this.url,
+    required this.type,
+    this.duration,
+    this.previewImageUrl,
+  });
+
+  /// Media key
+  ///
+  /// Originally retrieved by a tweet lookup and used to
+  /// retrieve more detailed Media data
+  final String key;
+
+  /// Media URL
+  final String url;
+
+  /// Type of media
+  ///
+  /// Possible values: photo,
+  final MediaType type;
+
+  /// Duration if the type is Video or GIF
+  final int? duration;
+
+  /// Static preview image for Video/GIF types
+  final String? previewImageUrl;
+}
```

Comment:
For me, the enum as well as the conditional properties look like an implementation using superclasses would be better. 

Therefore, we could have an abstract `Media` class which then implements an `ImageMedia`, `VideoMedia`, `GitMedia` class. 
What do you think?

Reply:
Talked about this. It's easier as it is right now.